### DISCUSSION TITLE
governing equation

### DISCUSSION THREAD

**Discussion by IlllIIIlllIl** [SynxFlow/SynxFlow#11] [Q&A] [answered]
Is the governing equation in debris.run and the corresponding file is debris.pyd? What should I do if I want to add a new term to the governing equation?

**Comment by xiaxilin** [comment, marked as answer]
Hi, to do this, you would need to change the simulation engine part, which is written by CUDA. Then the package needs to be recompiled. The entrance to the main function is [here](https://github.com/SynxFlow/SynxFlow/blob/master/synxflow/apps/cudaDebrisFlowSolverPybind/cuda_debris_flow_solver.cu)

**Reply by IlllIIIlllIl** [reply]
Hi,I hope I won't disturb your precious time. I am a beginner who just touches CFD models. I hope you can answer the following questions for me.1、Please check whether my understanding of the following parameters is correct. For the grid unit inside the fluid, h refers to the vertical distance between the top liquid surface of the fluid and the grid, z refers to the distance between the grid and the coordinate axis under the coordinate system, and eta_this refers to the height of the top liquid surface of the fluid in the coordinate system. If I understand correctly, is eta the same for structural grids, for grids in the same z direction?2、For hC, I see that you stated in the code that it is a solid volume parameter, then in a grid, does the parameter extracted by hC_this[index] refer to the solid volume in a grid cell?3、Is there a file that explains the meaning of all parameters, and I want to learn more about using parameters in your code from your code file?4、One last question, In your "A new depth-averaged model for Chaoow -like landslides over complex terrains with curvatures and steep slopes" article, See your semi-discretization of the governing equation, but no mention of the final algebraic equation, and your code is also based on this semi-discretization equation?
